### DISCUSSION TITLE
Where is the file app_code_configs.py?

### DISCUSSION THREAD

**Discussion by rlourette** [nasa/cFS#684] [Q&A]
Where is the official source for the file app_code_configs.py?

It is referenced in 
[https://github.com/nasa/CFS-101/blob/master/CFS-101-Guide/CFS-101-Guide/Html/CFS-101.html#_creating_the_new_applications_from_a_script](https://github.com/nasa/CFS-101/blob/master/CFS-101-Guide/CFS-101-Guide/Html/CFS-101.html#_creating_the_new_applications_from_a_script)
You have to extract the files from https://github.com/nasa/CFS-101/blob/master/CFS-101-Guide.7z to get to the HTML files.

Searching Github yields several locations:
[https://github.com/OpenSatKit/OpenSatKit/blob/132bb3971417e699acf0fd23f7df3bf4908e610f/cfs/tools/gen_app_code](https://github.com/OpenSatKit/OpenSatKit/blob/132bb3971417e699acf0fd23f7df3bf4908e610f/cfs/tools/gen_app_code)

[https://github.com/Spacecraft-Code/cFE/tree/77f9d0fbbe8b0cc9d92a70c2395912ac6c3ab319/cfe/tools/gen_app_code](https://github.com/Spacecraft-Code/cFE/tree/77f9d0fbbe8b0cc9d92a70c2395912ac6c3ab319/cfe/tools/gen_app_code)

**Comment by irowebbn** [comment]
I can only assume that it was removed sometime between the 6.4.0 and the 6.6 release, because the local copy I have has commits originating from the 6.4.0 release, and then 4 bugfix commits, the last of which was in May 2016. You can find these old releases at https://sourceforge.net/projects/coreflightexec/

My understanding is that this was a tool developed by a separate team a from the main cFS development group, at a different NASA center, but the two were packaged together for a while. The tool went unmaintained for some time, so it was removed. There have been multiple breaking API changes since that version, so any apps that the tool generates will not work out of the box.

**Comment by irowebbn** [comment]
Also, there's a related question already answered here: https://github.com/nasa/cFS/discussions/603
